### DISCUSSION TITLE
Language Feature?

### DISCUSSION THREAD

**Discussion by gafter** [dotnet/csharplang#2348] [Language Ideas]
@TheXenocide commented on [Thu Mar 14 2019](https://github.com/dotnet/roslyn/issues/34116)

I'm not sure what I would call this particular type of syntax sugar, but as I was reading through some source recently I wondered if there wasn't a readable way to simplify the following commonly occurring pattern? (Note: this is from some older sample source; I updated it here to use `out string value` already, where it was previously using `string value;` on the preceding line)

```csharp
return Items.TryGetValue(RedirectUriKey, out string value) ? value : null;
```
Surely it's not the most important feature request, but I figured it wouldn't hurt to ask. Maybe something like
```csharp
Items.TryGetValue(RedirectUriKey, return string value);
```
Allowing `return` to be used in place of `out` or `ref` (possibly even optimizing around allocation behavior? I'm not sure of the internals involved here). I honestly haven't thought super deeply about this so feel free to shoot it down, but it was a thought I figured I'd share.

---

@YairHalberstadt commented on [Thu Mar 14 2019](https://github.com/dotnet/roslyn/issues/34116#issuecomment-472981191)

This looks like it would be more appropriate for the csharplang repo, where requests for language changes are taken.

Also, if you're not sure of a suitable syntax, you might want to discuss your ideas on www.gitter.im/dotnet/csharplang first.

**Comment by HaloFour** [comment]
You could just do:

```cs
Items.TryGetValue(RedirectUriKey, out var value);
return value;
```

Ignoring the return value.  The `out` parameter is guaranteed to be assigned either way, and while you should confirm with the documentation generally `Try*` methods return the default value for that type if the action fails.

**Comment by TheXenocide** [comment]
True, and this is generally what I do (the sample above is actually from an MS AuthenticationProperties class related to their claims authentication abstraction). I don't suppose it would save much in that context, unless there's some sort of return register optimization to be had, which could probably be performed during compile-time optimization (if there was an optimization to be had here anyway) without the syntax sugar mentioned above.

**Comment by TheXenocide** [comment]
Also, thanks for the pointers @YairHalberstadt

**Comment by Thaina** [comment]
Normally I do

```C#
return Items.TryGetValue(RedirectUriKey, out var value) ? value : null;
```

I don't like your syntax. It hide the `return` in the middle of the line
